app/redux/reducers/localizationSlice.ts

```diff
@@ -1,96 +1,103 @@
-      state.luxonLocale = action.payload.language.luxonLocale;
-      state.locale = action.payload.language.locale;
-      state.pluralShow = action.payload.language.pluralShow;
-      }else {
-        console.log('country default is',action.payload)
-        state.countryId = action.payload.countryId;
-        state.languageCode = action.payload.languages[0].languageCode;
-        state.luxonLocale = action.payload.languages[0].luxonLocale;
-        state.locale = action.payload.languages[0].locale;
-        state.pluralShow = action.payload.languages[0].pluralShow;
+    onLocalizationSelect: (state, action: PayloadAction<any>): any => {
+      console.log('LanguageCode is', action.payload)
+      if (action?.payload?.language?.languageCode != undefined) {
+        state.countryId = action.payload.country.countryId;
+        state.languageCode = action.payload.language.languageCode;
+        state.luxonLocale = action.payload.language.luxonLocale;
+        state.locale = action.payload.language.locale;
+        state.pluralShow = action.payload.language.pluralShow;
+      } else {
+        console.log('country default is', action.payload)
+          state.countryId = action.payload.countryId;
+          state.countrySelectedId = action.payload.countryId;
+          state.languageCode = action.payload.languages[0].languageCode;
+          state.luxonLocale = action.payload.languages[0].luxonLocale;
+          state.locale = action.payload.languages[0].locale;
+          state.selectedLocale = action.payload.languages[0].locale;
+          state.pluralShow = action.payload.languages[0].pluralShow;
       }
     },
-    oncountrtIdChange: (state, action: PayloadAction<any>):any => {
-      state.countryId = action.payload;      
+    oncountrtIdChange: (state, action: PayloadAction<any>): any => {
+      state.countryId = action.payload;
     },
-    
+
     setSponsorStore: (
       state,
       action: PayloadAction<any>,
-    ):any => {
+    ): any => {
       state.sponsors = action.payload;
     },
     setChildStore: (
       state,
       action: PayloadAction<any>,
-    ):any => {
-        console.log(action.payload,"--",state);
+    ): any => {
+      console.log(action.payload, "--", state);
     },
     setrestartOnLangChange: (
       state,
       action: PayloadAction<any>,
-    ):any => {
+    ): any => {
       state.restartOnLangChange = action.payload;
     },
     setAppLayoutDirection: (
       state,
       action: PayloadAction<any>,
-    ):any => {
+    ): any => {
       state.AppLayoutDirection = action.payload;
     },
     setAppLayoutDirectionScreen: (
       state,
       action: PayloadAction<any>,
-    ):any => {
+    ): any => {
       state.AppLayoutDirectionScreen = action.payload;
     },
     setAppLayoutDirectionParams: (
       state,
       action: PayloadAction<any>,
-    ):any => {
-      console.log('Hiiiiiiiiii',action.payload);
+    ): any => {
+      console.log('Hiiiiiiiiii', action.payload);

```

Comment:
You can remove or improve dummy comments.

Reply:
yes @Akhror I will remove all unnecessary comments